libnetwork/iptables/firewalld.go

```diff
@@ -52,11 +60,46 @@ func firewalldInit() error {
 	if connection, err = newConnection(); err != nil {
 		return fmt.Errorf("Failed to connect to D-Bus system bus: %v", err)
 	}
-	firewalldRunning = checkRunning()
+
+	// If firewalld is not running on the first check poll
 	if !firewalldRunning {
+		ctx, cancel := context.WithTimeout(context.Background(), fireWalldInitializationTimeout)
+		defer cancel()
+
+		for {
+
+			cctx, ccancel := context.WithTimeout(ctx, dbusConnectionTimeout)
+			firewalldRunning, err = func(ctx context.Context) (bool, error) {
+
+				var running bool
+				log.G(ctx).Debugf("Checking status of Firewalld")
+				running, err = checkRunning(ctx)
+
+				if err != nil {
+					return running, err
+				}
+				return running, nil
+			}(cctx)
+			ccancel()
+			if firewalldRunning || err != nil {
+				break
+			}
+			time.Sleep(dbusConnectionTimeout)
+		}
+
+		if err != nil {
+			log.G(context.TODO()).Debugf("Check running failed: %v", err)
+		}
+	}
+
+	if !firewalldRunning {
+		log.G(context.TODO()).Debugf("The dbus API is accessible but firewalld is currently unavailable continuing.")
 		connection.sysconn.Close()
 		connection = nil
```

Comment:
If firewalld is stuck in `INIT`, I think `firewalldRunning` will be false and `err` will be a DeadlineExceeded ... so, firewalld won't be configured. From comments on the other review, that may not be the intention?

So, this code might need to distinguish between:
- the 2.5s timeout being reached in `INIT`, perhaps to be treated as firewalld-running
- dbus not responding within the 1s timeout, meaning firewalld/dbus might really be broken - that probably shouldn't be treated as firewalld-running, because it probably wouldn't have been before.
- dbus returning an error that says it doesn't know anything about firewalld

(Either way, it'd be worth adding a comment to explain how `INIT` is handled. The log message could also be tidied up - the "continuing" isn't needed but, if it's staying, it needs a comma or something.)